File: client/src/application/class_sections/dto/teacher_class_section.dto.js
class TeacherClassSectionDTO {

  constructor({
    university_id,
    department_id,

    class_section_id,

    academic_year_id,
    academic_year_name,

    semester_id,
    semester_name,
    semester_number,

    subject_id,
    subject_name,

    group_number,

    status,
  }) {

    this.universityId = university_id;
    this.departmentId = department_id;

    this.classSectionId = class_section_id;

    this.academicYearId = academic_year_id;
    this.academicYearName = academic_year_name;

    this.semesterId = semester_id;
    this.semesterName = semester_name;
    this.semesterNumber = semester_number;

    this.subjectId = subject_id;
    this.subjectName = subject_name;

    this.groupNumber = group_number;

    this.status = status;
  }


  static fromResponse(data) {

    return new TeacherClassSectionDTO(data);
  }

}


module.exports = TeacherClassSectionDTO;

File: server/apps/class_sections/application/dto/teacher_class_section_dto.py
from dataclasses import dataclass


@dataclass
class TeacherClassSectionDTO:
    university_id: str
    department_id: str

    class_section_id: str

    academic_year_id: str
    academic_year_name: str

    semester_id: str
    semester_name: str
    semester_number: str

    subject_id: str
    subject_name: str

    group_number: int
    status: str